src/mission_ui.cpp

```diff
@@ -71,9 +71,13 @@ class mission_ui_impl : public cataimgui::window
         mission_ui_tab_enum selected_tab = mission_ui_tab_enum::ACTIVE;
         mission_ui_tab_enum switch_tab = mission_ui_tab_enum::num_tabs;
 
-        size_t window_width = str_width_to_pixels( TERMX ) / 2;
-        size_t window_height = str_height_to_pixels( TERMY ) / 2;
-        size_t table_column_width = window_width / 2;
+        float window_width = std::clamp( ImGui::GetMainViewport()->Size.x / 2,
+                                         float( str_width_to_pixels( 80 ) ),
+                                         ImGui::GetMainViewport()->Size.x );
+        float window_height = std::clamp( ImGui::GetMainViewport()->Size.y / 2,
+                                          float( str_height_to_pixels( 24 ) ),
+                                          ImGui::GetMainViewport()->Size.y );
```

Comment:
```suggestion
        float window_width = std::clamp( float( str_width_to_pixels( 80 ) ),
                                         ImGui::GetMainViewport()->Size.x / 2,
                                         ImGui::GetMainViewport()->Size.x );
        float window_height = std::clamp( float( str_height_to_pixels( 24 ) ),
                                          ImGui::GetMainViewport()->Size.y / 2,
                                          ImGui::GetMainViewport()->Size.y );
```

The other way around the upper limit of the clamp would be meaningless (viewport is always larger than half viewport), and if 80x24 terminal would for some reason be larger than viewport size (although it shouldn't be able to), it'd be undefined behavior according to cppreference.

Reply:
Works and passes test, seems good. All the same color in the end to me 👍